stablehlo/dialect/Base.td

```diff
@@ -248,8 +255,8 @@ def HLO_StaticDimensionTensor : RankedTensorOf<[HLO_DimensionValue], [HasStaticS
 def HLO_StaticShapeTensor : StaticShapeTensorOf<[
       HLO_Float, HLO_Pred, HLO_Int, HLO_Complex, HLO_QuantizedInt]>;
 
-def HLO_StaticShapeTensorOrPerAxisQuantizedTensor : StaticShapeTensorOf<[
-      HLO_Float, HLO_Pred, HLO_Int, HLO_Complex, HLO_QuantizedInt, HLO_PerAxisQuantizedInt]>;
+def HLO_StaticShapeTensorOrPerAxisQuantizedTensor : RankedTensorOf<[HLO_Float, HLO_Pred, HLO_Int, HLO_Complex, HLO_QuantizedInt, HLO_PerAxisQuantizedInt],
```

Comment:
Initially we had
```
def HLO_StaticShapeTensorOrPerAxisQuantizedTensor : StaticShapeTensorOf<[
      HLO_Float, HLO_Pred, HLO_Int, HLO_Complex, HLO_QuantizedInt, HLO_PerAxisQuantizedInt]>;
```

`StaticShapeTensprOf` [cs](https://github.com/llvm/llvm-project/blob/c515c780244e3ecbb1fcfd06b3ad588d8d22c28e/mlir/include/mlir/IR/CommonTypeConstraints.td#L712) does not expose the pred which we would need to pass on `IsValidQuantizedDimension`.

This is same issue with https://github.com/openxla/stablehlo/blob/1beb2c3ad9a0650477367dabd081d232b771352e/stablehlo/dialect/Base.td#L244 resolved similarly.

One option we explored is to use something like
```
AnyTypeOf[
    StaticShapeTensorOf<[HLO_Float, HLO_Pred, HLO_Int, HLO_Complex, HLO_QuantizedInt]>,
   RankedTensorOf<[HLO_PerAxisQuantizedInt],
    [IsValidQuantizedDimension, HasStaticShapePred]>;
]
```

But that would require explicit casting in the code base at several places because  the indirection `AnyTypeOf` does not allow the type to be recognized as RankedTensor (despite the fact that  both of its constituents are RankedTensors)

Reply:
makes sense, didn't notice the `HasStaticShapePred` constraint initially.